src/adapters/models/t5/modeling_t5.py

```diff
@@ -144,7 +144,10 @@ def project(hidden_states, proj_layer, key_value_states, past_key_value):
             query_states, key_states.transpose(3, 2)
         )  # equivalent of torch.einsum("bnqd,bnkd->bnqk", query_states, key_states), compatible with onnx op>9
 
-        if position_bias is None:
+        # For Prefix Tuning, when training with AdapterDrop, we must additionally check if the dimensions of the
+        # position_bias given from previous layers match the dimensions of the scores of the current layer to make
+        # sure that the position_bias is adequately recomputed if previous layers have been skipped.
+        if position_bias is None or position_bias.shape != scores.shape:
```

Comment:
can we specifically check the seqlen dimensions here to make this condition more precise?